std/algorithm/comparison.d

```diff
@@ -2271,13 +2271,13 @@ if (alternatives.length >= 1 &&
     auto c = 2;
     const d = 3;
     auto cd = either!(a => a == 3)(c, d); // use predicate
-    static assert(is(typeof(cd) == int));
+    //static assert(is(typeof(cd) == const(int)));
```

Comment:
Why are these all suddenly becoming `const`? And simply commenting out tests is definitely not good. Are you planning to comment them out now and then create a separate PR to uncomment them once the dmd PR is merged? Either way, it sounds like this is a breaking change. Having a bunch of ranges suddenly have const elements is definitely going to break code - as evidenced by the change to std.datetime in this PR. What's the motivation here? Looking at the bugs referenced in the dmd PR, they all refer to widening types, not changing to `const`.

Reply:
Yes, the PR was mentioned in the initial comment https://github.com/dlang/dmd/pull/9943.
See also this discussion with Walter prior to this https://github.com/dlang/dmd/pull/9893#issuecomment-496031539.

Reply:
The loss of `const` was linked to the widening.
What do you think the type of `[0, cast(const)1]` should be? `const(int)[]` or `int[]`, likewise with `[cast(const)1]`.

Reply:
This is not clearly defined in the spec yet.

Reply:
If we weren't talking about types which implicitly converted from `const` to mutable, the resulting elements would need to be `const` (and thus have to be now without your changes), but when they implicitly convert to mutable, having them be mutable is far more useful and wouldn't break code the way that the current changes are. If @WalterBright is in favor of making the result `const`, then fine, but it's clearly going to break code, and in general is going to be annoying. `[cast(const)1]` by itself should obviously be `const(int)[]` though, since all of the elements are `const` (and that's the current behavior).
Either way, the change needs to be approved by @WalterBright , and I see no discussion of arrays in the linked PRs.

Reply:
Arrays were affected because DMD uses a ternary expression (`?:`) internally to find a common type for array literals. As It does it with associative array literals (`[cast(const)"0": 0]` is `int[const(string)]`)

Reply:
The way that the compiler decides the type of an array literal has never seemed consistent to me, and this change definitely seems to be for the worse with regards to `const`. Regardless, the dmd PR needs to be approved before it makes any sense to merge this one. And if this PR gets merged, another one will need to be created where the tests are not commented out.